Remotion/Data/DomainObjects/Infrastructure/Serialization/FlattenedSerializationInfo.cs

```diff
@@ -54,7 +54,9 @@ public void AddBoolValue (bool value)
     /// <remarks>Note: This will use T to decide whether to recurse into an IFlattenedSerializable or not.</remarks>
     public void AddValue<T> (T? value)
     {
-      if (typeof(IFlattenedSerializable).IsAssignableFrom(typeof(T)))
+      if (typeof(T) == typeof(Type))
+        AddHandle((((Type?)(object?)value)?.AssemblyQualifiedName));
```

Comment:
Same as with the serializer: need to factor out the type-string conversion. Yes, it's just one property, but there's different properties and which property we use is part of the business logic that should not be duplicated.